### PR TITLE
Allow modifying anchors with anchor(subtotal)<- syntax

### PR DESCRIPTION
**Pull request by jonkeane**
Refactored some `PATCH`es to use more package-standard approaches.

### CODE REVIEW

File: tests/testthat/test-subtotals-and-headings.R [hunk 1]
```diff
@@ -184,4 +234,47 @@ with_test_authentication({
...
+        expect_equal(names(subtotals(ds$v4)), c("This is a subtitle", "B alone",
+                                                "C alone", "B+C"))
+        expect_equal(anchors(subtotals(ds$v4)), c("top", 1, 2, "bottom"))
+        expect_equal(arguments(subtotals(ds$v4)[[1]]), NA)
+        expect_equal(arguments(subtotals(ds$v4)[[2]]), 1)
+        expect_equal(arguments(subtotals(ds$v4)[[3]]), 2)
+        expect_equal(arguments(subtotals(ds$v4)[[4]]), c(1,2))
+
+        # changing names
+        expect_silent(name(subtotals(ds$v4)[[1]]) <- "The new subtitle")
```
Comment (nealrichardson) [inline tests/testthat/test-subtotals-and-headings.R]: why all the `expect_silent`? seems to mask (from my readability) the operations you're doing that are changing state, for little value.

File: R/subtotals-and-headings.R [hunk 2]
```diff
@@ -218,18 +218,10 @@ setMethod("subtotals<-", c("CrunchVariable", "ANY"), function (x, value) {
...
-    if (!is.null(old_inserts)) {
-        inserts <- modifyCats(old_inserts, inserts)
-    }
-
-    bd <- wrapEntity("view" = list("transform" = list("insertions" = inserts)))
-    crPATCH(self(x), body=toJSON(bd))
     dropCache(cubeURL(x))
+    bd <- list("transform" = list("insertions" = inserts))
+    # setEntitySlot manages old inserts so they are not duplicated
+    ent <- setEntitySlot(entity(x), "view", bd)
```
Comment (nealrichardson) [inline R/subtotals-and-headings.R]: I'd put the `dropCache` statements after doing the PATCH (`setEntitySlot`)

File: tests/testthat/test-subtotals-and-headings.R (hunk 1 above)
Comment (jonkeane) [reply tests/testthat/test-subtotals-and-headings.R]: Old habits — I've taken them out.

File: R/subtotals-and-headings.R (hunk 2 above)
Comment (jonkeane) [reply R/subtotals-and-headings.R]: That makes sense (and my first intuition as well), I was modeling this after https://github.com/Crunch-io/rcrunch/blob/master/R/variable.R#L115-L116, which I've changed to drop the cache after as well.

Comment (nealrichardson) [reply R/subtotals-and-headings.R]: Good call, thanks.

**Pull request merged**